invenio_indexer/cli.py

```diff
@@ -52,3 +53,19 @@ def run(delayed, concurrency):
     else:
         click.secho("Indexing records...", fg='green')
         RecordIndexer().process_bulk_queue()
+
+
+@index.command()
+@click.argument('pid_type')
+@click.option('-i', '--index', default=None, help="the index name.")
+@click.option('-d', '--doctype', default=None, help="the doc_type.")
+@with_appcontext
+def reindex(pid_type, index, doctype):
+    """Reindex all records."""
+    query = (x[0] for x in PersistentIdentifier.query.filter_by(
+        pid_type=pid_type, object_type='rec'
+    ).values(
+        PersistentIdentifier.object_uuid
+    ))
+    click.echo("Sending tasks...")
+    RecordIndexer().bulk_index(query, index=index, doc_type=doctype)
```

Comment:
It's correct what @hachreak is doing. The job of the CLI is only to request indexing of the given records, not to actually index them. The indexing should happen in separate celery tasks. Which you can start like this:

``<cmd> index -d --concurrency 8``

Having multiple indexing tasks running at once will get you much higher ingestion rates that what a single process can do alone.